src/bun.js/http.exports.js

```diff
@@ -40,16 +40,85 @@ var _globalAgent;
 var _defaultHTTPSAgent;
 var kInternalRequest = Symbol("kInternalRequest");
 
-var FakeSocket = class Socket {
-  on() {
+var FakeSocket = class Socket extends Duplex {
+  bytesRead = 0;
+  bytesWritten = 0;
+  connecting = false;
+  remoteAddress = null;
+  localAddress = "127.0.0.1";
+  remotePort;
+  timeout = 0;
+
+  isServer = false;
+
+  address() {
+    return {
+      address: this.localAddress,
+      family: this.localFamily,
+      port: this.localPort,
+    };
+  }
+
+  get bufferSize() {
+    return this.writableLength;
+  }
+
+  connect(port, host, connectListener) {
+    return this;
+  }
+
+  _destroy(err, callback) {}
+
+  _final(callback) {}
+
+  get localAddress() {
+    return "127.0.0.1";
+  }
+
+  get localFamily() {
+    return "IPv4";
+  }
+
+  get localPort() {
+    return 80;
+  }
+
+  get pending() {
+    return this.connecting;
+  }
+
+  _read(size) {}
+
+  get readyState() {
+    if (this.connecting) return "opening";
+    if (this.readable) {
+      return this.writable ? "open" : "readOnly";
+    } else {
+      return this.writable ? "writeOnly" : "closed";
+    }
+  }
+
+  ref() {}
+
+  get remoteFamily() {
+    return "IPv4";
+  }
+
+  resetAndDestroy() {}
+
+  setKeepAlive(enable = false, initialDelay = 0) {}
+
+  setNoDelay(noDelay = true) {
     return this;
   }
-  off() {}
-  addListener() {
+
+  setTimeout(timeout, callback) {
     return this;
   }
-  removeListener() {}
-  removeAllListeners() {}
+
+  unref() {}
+
+  _write(chunk, encoding, callback) {}
```

Comment:
is this return type correct

Reply:
Yes on our `node:net` this does not return anything